diff --git a/ui-shared.c b/ui-shared.c
index 72a1505..45b8524 100644
--- a/ui-shared.c
+++ b/ui-shared.c
@@ -1002,8 +1002,6 @@ static void print_header(void)
 
 	html("<td class='main'>");
 	if (ctx.repo) {
-		cgit_index_link("index", NULL, NULL, NULL, NULL, 0, 1);
-		html(" : ");
 		cgit_summary_link(ctx.repo->name, NULL, NULL, NULL);
 		if (ctx.env.authenticated) {
 			html("</td><td class='form'>");
